exchanges/gateio/gateio_websocket.go

```diff
@@ -764,11 +739,157 @@ func singleSymbolChannel(name string) bool {
+	switch a {
+	case asset.Spot:
+		switch channelName {
+		case spotOrderbookTickerChannel:
+			if interval != kline.TenMilliseconds {
+				return fmt.Errorf("%w for `%s` only `%s` is supported", subscription.ErrInvalidInterval, channel, kline.TenMilliseconds)
+			}
+			if levels != 1 {
+				return fmt.Errorf("%w for `%s` only `1` is supported", subscription.ErrInvalidLevel, channel)
+			}
+		case spotOrderbookUpdateChannel:
+			if interval != kline.HundredMilliseconds {
+				return fmt.Errorf("%w for `%s` only `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds)
+			}
+			if levels != 100 {
+				return fmt.Errorf("%w for `%s` only `%d` is supported", subscription.ErrInvalidLevel, channel, 100)
+			}
+		case spotOrderbookChannel:
+			if interval != kline.HundredMilliseconds && interval != kline.ThousandMilliseconds {
+				return fmt.Errorf("%w for `%s` only `%s` or `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds, kline.ThousandMilliseconds)
+			}
+			if levels != 5 && levels != 10 && levels != 20 && levels != 50 && levels != 100 {
+				return fmt.Errorf("%w for `%s` only `5`, `10`, `20`, `50` or `100` is supported", subscription.ErrInvalidLevel, channel)
+			}
+		}
+	case asset.Futures, asset.DeliveryFutures:
+		switch channelName {
+		case futuresOrderbookChannel:
+			if interval != 0 {
+				return fmt.Errorf("%w for `%s` only `0` is supported", subscription.ErrInvalidInterval, channel)
+			}
+			if levels != 1 && levels != 5 && levels != 10 && levels != 20 && levels != 50 && levels != 100 {
+				return fmt.Errorf("%w for `%s` only `1`, `5`, `10`, `20`, `50` or `100` is supported", subscription.ErrInvalidLevel, channel)
+			}
+		case futuresOrderbookTickerChannel:
+			if interval != 0 {
+				return fmt.Errorf("%w for `%s` only `0` is supported", subscription.ErrInvalidInterval, channel)
+			}
+			if levels != 1 {
+				return fmt.Errorf("%w for `%s` only `1` is supported", subscription.ErrInvalidLevel, channel)
+			}
+		case futuresOrderbookUpdateChannel:
+			if a == asset.Futures {
+				if levels != 20 && levels != 50 && levels != 100 {
+					return fmt.Errorf("%w for `%s` only `20`, `50` or `100` is supported", subscription.ErrInvalidLevel, channel)
+				}
+				if levels != 20 {
+					if interval == kline.TwentyMilliseconds {
+						return fmt.Errorf("%w for `%s` only `20` levels supports `%s`", subscription.ErrInvalidInterval, channel, kline.TwentyMilliseconds)
+					}
+					if interval != kline.HundredMilliseconds {
+						return fmt.Errorf("%w for `%s` only `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds)
+					}
+				} else if interval != kline.HundredMilliseconds && interval != kline.TwentyMilliseconds {
+					return fmt.Errorf("%w for `%s` only `%s` or `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds, kline.TwentyMilliseconds)
+				}
```

Comment:
I'm confused by this line, because ... the template won't output an interval for `futuresOrderbookUpdateChannel`, will it?

Reply:
No, its ready for when templates do futures as well, so it will scale up without having to worry about it.